client/my-sites/checkout/composite-checkout/hooks/use-is-apple-pay-available.ts

```diff
@@ -1,133 +0,0 @@
-/**
- * External dependencies
- */
-import { useEffect, useState } from 'react';
-import type { Stripe, StripeConfiguration } from '@automattic/calypso-stripe';
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:composite-checkout:use-is-apple-pay-available' );
-
-declare global {
-	interface Window {
-		ApplePaySession?: ApplePaySessionInterface;
-	}
-}
-
-interface ApplePaySessionInterface {
-	canMakePayments: () => boolean;
-}
-
-export interface CanMakePaymentState {
-	isLoading: boolean;
-	canMakePayment: boolean;
-}
-
-export default function useIsApplePayAvailable(
-	stripe: Stripe | null,
-	stripeConfiguration: StripeConfiguration | null,
-	isStripeError: boolean,
-	currency: string | null
-): CanMakePaymentState {
-	const [ canMakePayment, setCanMakePayment ] = useState< CanMakePaymentState >( {
-		isLoading: true,
-		canMakePayment: false,
-	} );
-
-	useEffect( () => {
-		let isSubscribed = true;
-		const unsubscribe = () => {
-			isSubscribed = false;
-		};
-
-		// Only calculate this once
-		if ( ! canMakePayment.isLoading ) {
-			return unsubscribe;
-		}
-
-		// If stripe did not load, we will never load
-		if ( isStripeError ) {
-			debug( 'isApplePayAvailable giving up due to stripe error' );
-			setCanMakePayment( { isLoading: false, canMakePayment: false } );
-			return unsubscribe;
-		}
-
-		// We'll need the Stripe library so wait until it is loaded
-		if ( ! stripe || ! stripeConfiguration ) {
-			debug( 'isApplePayAvailable waiting on stripe' );
-			return unsubscribe;
-		}
-
-		// Our Apple Pay implementation uses the Payment Request API, so
-		// check that first.
-		if ( ! window.PaymentRequest ) {
-			debug( 'isApplePayAvailable giving up because there is no paymentRequest API' );
-			setCanMakePayment( { isLoading: false, canMakePayment: false } );
-			return unsubscribe;
-		}
-
-		// Ask the browser if apple pay can be used. This can be very
-		// expensive on certain Safari versions due to a bug
-		// (https://trac.webkit.org/changeset/243447/webkit)
```

Comment:
It wasn't completely obvious to me why it's OK to remove this section.  Is the same thing being checked somewhere else?

Reply:
Oh, good question. I had copied that check from old checkout, but when I went and [looked up the bug in question](https://bugs.webkit.org/show_bug.cgi?id=196179), it's been fixed since 2019-03-25. 

In addition, since we now use Stripe's code to test if Apple Pay is available, rather than going to the browser itself, I believe we can trust it to work on all browsers.

(The main reason for removing it was because this function will now check for both Apple Pay and Google Pay, although we could have refactored to keep the logic, because of the above I decided it was probably no longer needed.)